### PR TITLE
Sketch feature

### PR DESCRIPTION
**Pull request by PedroRodP**
Hey! I liked your plugin and really wanted to use the sketch mode, but since D2 parameters are not open to edit through an input field (this might be a nice option that allow people to play with D2 without major efforts) I forked your project and made this little enhancement. Hope it can be included in the original plugin!

### CODE REVIEW

**Comment by PedroRodP** [comment]
**Plugin with feature toggled off**
image

**Plugin with feature toggled on**
image

File: src/resources/META-INF/plugin.xml [hunk 1]
```diff
@@ -53,6 +53,9 @@
            class="org.jetbrains.plugins.d2.action.D2LayoutEngineActionGroup"
            popup="true"
            icon="AllIcons.General.Layout"/>
+    <action id="D2.SketchMode"
+           class="org.jetbrains.plugins.d2.action.D2SketchToggleAction"
+           icon="AllIcons.Actions.EnableNewUi"/>
```
Comment (goto1134) [inline src/resources/META-INF/plugin.xml:58]: Let's use another icon here. Something to highlight is that the style looks hand-drawn. A random option:

image

File: src/resources/messages/D2Bundle.properties [hunk 2]
```diff
@@ -21,6 +21,9 @@ group.D2.ThemeChooser.description=Choose a theme for D2 diagram
 group.D2.LayoutChooser.text=Layout Chooser
 group.D2.LayoutChooser.description=Choose a layout for D2 diagram
 
+action.D2.SketchMode.text=Sketch
```
Comment (goto1134) [inline src/resources/messages/D2Bundle.properties:24]: Should it be `Sketch Mode` instead?

File: src/src/editor/D2CompilerState.kt [hunk 3]
```diff
@@ -13,6 +13,7 @@ internal class D2CompilerState(
   val port: Int,
   val theme: String?,
   val layout: D2Layout?,
+  val sketch: Boolean?,
```
Comment (goto1134) [inline src/src/editor/D2CompilerState.kt:16]: Can we switch to a nonnull type `Boolean` here?

File: src/src/editor/D2Viewer.kt [hunk 4]
```diff
@@ -45,7 +45,9 @@ private class ProjectLevelCoroutineScopeHolder(val coroutineScope: CoroutineScop
 internal const val D2_EDITOR_NAME = "D2FileEditor"
 
 @Serializable
-internal data class D2FileEditorState(@JvmField var theme: D2Theme?, @JvmField val layout: D2Layout?) : FileEditorState {
+internal data class D2FileEditorState(@JvmField var theme: D2Theme?,
+                                      @JvmField val layout: D2Layout?,
+                                      @JvmField val sketch: Boolean?) : FileEditorState {
```
Comment (goto1134) [inline src/src/editor/D2Viewer.kt:50]: We could specify a default value here with `@Default(false)` and make the field not nullable

File: src/resources/META-INF/plugin.xml (hunk 1 above)
Comment (PedroRodP) [reply src/resources/META-INF/plugin.xml:58]: Could you please provide this icon ID? I'm not sure I've seen it within the available IntelliJ options

File: src/resources/messages/D2Bundle.properties (hunk 2 above)
Comment (PedroRodP) [reply src/resources/messages/D2Bundle.properties:24]: Ok, sure

File: src/src/editor/D2CompilerState.kt (hunk 3 above)
Comment (PedroRodP) [reply src/src/editor/D2CompilerState.kt:16]: I'll give it a try. This is my first time with Kotlin 😄

File: src/resources/META-INF/plugin.xml (hunk 1 above)
Comment (goto1134) [reply src/resources/META-INF/plugin.xml:58]: Sorry, I can't provide you with a specific id. Please, take a look at the existing icons here: https://jetbrains.design/intellij/resources/icons_list/

If there are no suitable icons, consider adding one following the documentation here: https://plugins.jetbrains.com/docs/intellij/icons.html

Comment (goto1134) [reply src/resources/META-INF/plugin.xml:58]: Thank you for your research on that! For me, the pencil looks like the best option here :)

File: src/src/editor/D2Viewer.kt (hunk 4 above)
Comment (PedroRodP) [reply src/src/editor/D2Viewer.kt]: Couldn't find a suitable @Default annotation, set the default with "= false"

File: src/src/editor/D2CompilerState.kt (hunk 3 above)
Comment (PedroRodP) [reply src/src/editor/D2CompilerState.kt]: Achieved 😄

File: src/resources/messages/D2Bundle.properties (hunk 2 above)
Comment (PedroRodP) [reply src/resources/messages/D2Bundle.properties]: Changed

File: src/resources/META-INF/plugin.xml (hunk 1 above)
Comment (PedroRodP) [reply src/resources/META-INF/plugin.xml]: Uploaded the first one for now

Comment (PedroRodP) [reply src/resources/META-INF/plugin.xml]: Sorry, which one then?

Option 1 (already uploaded): 
image


Option 2:
image


Option 3 (yours, has license):
image

Comment (PedroRodP) [reply src/resources/META-INF/plugin.xml]: Great. All set then. Thanks for your feedback!

File: src/src/editor/D2Viewer.kt (hunk 4 above)
Comment (goto1134) [reply src/src/editor/D2Viewer.kt]: I'm not familiar with kotlinx.serialization enough. Sorry for a misleading suggestion :)

The default value looks like the right solution according to the docs.

**Comment by develar** [comment]
Sorry for a long delay :)

**Pull request merged**